lib/rubocop/cop/rspec/hooks_before_examples.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for before/around/after hooks that come after an example.
+      #
+      # @example
+      #   # Bad
+      #
+      #   it 'checks what foo does' do
+      #     expect(foo).to be
+      #   end
+      #
+      #   before { prepare }
+      #   after { clean_up }
+      #
+      #   # Good
+      #   before { prepare }
+      #   after { clean_up }
+      #
+      #   it 'checks what foo does' do
+      #     expect(foo).to be
+      #   end
+      #
+      class HooksBeforeExamples < Cop
+        include RangeHelp
+        include RuboCop::RSpec::FinalEndLocation
+
+        MSG = 'Move `%<hook>s` before the examples in the group.'.freeze
+
+        def_node_matcher :hook?, Hooks::ALL.block_pattern
```

Comment:
Is this necessary? We have it defined here https://github.com/rubocop-hq/rubocop-rspec/blob/master/lib/rubocop/rspec/example_group.rb#L20

Reply:
Well, this will require instantiating ExampleGroup. But since this matcher is common, we can move it to RSpec::Language::NodePattern module

Reply:
I've extracted them in #659 and will update this if 659 is merged.

10x @geniou for mentioning this unnecessary duplication